src/ctapipe/io/dl2_tables_preprocessing.py

```diff
@@ -349,29 +351,52 @@ def get_simulation_information(
         if not has_pyirf:
             raise OptionalDependencyMissing("pyirf")
 
-        sim = loader.read_simulation_configuration()
+        sim_config = loader.read_simulation_configuration()
+        n_showers: int = 0
         try:
-            show = loader.read_shower_distribution()
+            shower_distribution = loader.read_shower_distribution()
+            n_showers = shower_distribution["n_entries"].sum()
         except NoSuchNodeError:
             self.log.warning(
-                "Simulation distributions were not found in the input files, falling back to estimating the number of showers from the simulation configuration."
+                "Simulation distributions were not found in the input files, "
+                "falling back to estimating the number of showers from the "
+                "simulation configuration."
             )
-            show = Table([sim["n_showers"]], names=["n_entries"], dtype=[np.int64])
 
-        for itm in ["spectral_index", "energy_range_min", "energy_range_max"]:
-            if len(np.unique(sim[itm])) > 1:
+        # Some tight consistency checks. Eventually we will support using the
+        # arbitrary shower distribution and non-flat spatial distributions.
+        # Currently we do not support those, so we raise exceptions here to
+        # avoid that we incorrectly compute the effective area, which would have
+        # a high scientific impact.
+        for itm in [
+            "spectral_index",
+            "energy_range_min",
+            "energy_range_max",
+            "max_scatter_range",
+            "max_viewcone_radius",
+            "min_viewcone_radius",
+        ]:
+            if len(np.unique(sim_config[itm])) > 1:
                 raise NotImplementedError(
                     f"Unsupported: '{itm}' differs across simulation runs"
                 )
 
+        n_showers_config = sim_config["n_showers"].sum() * sim_config["shower_reuse"][0]
```

Comment:
```suggestion
        n_showers_config = (sim_config["n_showers"] * sim_config["shower_reuse"]).sum()
```

otherwise it will be wrong for individual re-use. Or check reuse above, but here it's easy to do the right thing.